Distribution initial test

kmilstein opened the pull request:
Created a interface for EmailService and changed the name of it to
EmailServiceImpl. Changed setEmailServiceMethod in EmailScheduler to
reflect name change. Created DistributionController and DistributionService
classes. Created tests for EmailService and DistributionController.

kmilstein committed:
Created Initial Tests

Created a interface for EmailService and changed the name of it to
EmailServiceImpl. Changed setEmailServiceMethod in EmailScheduler to
reflect name change. Created DistributionController and DistributionService
classes. Created tests for EmailService and DistributionController.

a-holliday committed:
Add documentation to Distribution Controller

a-holliday committed:
Add second post mapping and tests for distribution controller

kmilstein committed:
Bug fix in DistributionControllerTest

Missing parameters were added to the tests for sendEmailsByCSV methods and
added documentation.

jurczakn commented on SurveyService/src/main/java/com/revature/controller/DistributionController.java:
```diff
@@ -0,0 +1,43 @@
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller takes in a batch id, or a batch id with a set of emails, then
```
Place validation description in Service where it belongs.  Have this specifically explain purpose of rest endpoints.

jurczakn commented on SurveyService/src/main/java/com/revature/controller/DistributionController.java:
```diff
@@ -0,0 +1,43 @@
+
+		return null;
+	}
+	
```
not documented

jurczakn commented on SurveyService/src/main/java/com/revature/service/DistributionServiceImpl.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
```
no documentation

jurczakn commented on SurveyService/src/main/java/com/revature/service/EmailServiceImpl.java:
```diff
@@ -0,0 +1,97 @@
+	 * @param destination email address
+	 * @throws AddressException exception thrown when a wrongly formatted address is encountered.
+	 * @throws MessagingException base class for all exceptions thrown by the Messaging classes
+	 */
```
update method name to send email singular

jurczakn commented on SurveyService/src/main/java/com/revature/service/EmailService.java:
```diff
@@ -1,97 +1,10 @@
-	@Value("${survey_service.email_service.password}")
-	private String emailServicePassword;
+public interface EmailService {
 	
```
change to send email singular

jurczakn commented on SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:
```diff
@@ -0,0 +1,151 @@
+		Mockito.when(service.sendEmailsByBatchId(validBatchId)).thenReturn(invalidEmails);
+
+		RequestBuilder request = MockMvcRequestBuilders.post("/distribute/" + validBatchId);
+		MvcResult result = mockMvc.perform(request).andReturn();
```
add verification service is called

jurczakn commented on SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:
```diff
@@ -0,0 +1,151 @@
+		Mockito.when(service.sendEmailsByBatchId(validBatchId)).thenReturn(invalidEmails);
+
+		RequestBuilder request = MockMvcRequestBuilders.post("/distribute/" + validBatchId);
+		MvcResult result = mockMvc.perform(request).andReturn();
```
check status code as well

jurczakn commented on SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:
```diff
@@ -0,0 +1,151 @@
+ */
+@SpringBootTest
+@AutoConfigureMockMvc
+@WebMvcTest
```
test for improperly formatted .csv

kmilstein replied:
Test has been added

jurczakn commented on SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:
```diff
@@ -0,0 +1,151 @@
+		assertEquals("", result.getResponse().getContentAsString());
+
+	}
+
```
comment test

kmilstein replied:
Commends have been added

jurczakn reviewed (changes requested):
update, and notify when fixed

leonardoz15 committed:
bug fix sendEmailsByCSV in distribution controller

removed trailing '/' in sendEmailsByCSV endpoint mapping within the distribution controller. Also removed the trailing slash in controller tests.

leonardoz15 committed:
big fix: code review fixes

Renamed sendEmails to sendEmail throught application. Fleshed out documentation for distribution controller, service and controller test. Changed param in distribution controller test from file.csv to emails.csv. Added a placeholder for distributionControllerSendEmailsByBatchId_withInvalidCSV test case.

kmilstein committed:
Fixed DistributionController Test

Upload all references from File to MultipartFile in classes, related to the
distribution controller. Additional the method call in MockMvcRequestBuilder
was changed from .post() to .multipart()

kmilstein commented:
Requested changes have been made.

jurczakn commented on SurveyService/src/main/java/com/revature/controller/DistributionController.java:
```diff
@@ -0,0 +1,46 @@
+	 */
+	@PostMapping("/distribute/{batchId}")
+	private ResponseEntity<List<String>> sendEmailsByBatchId(@PathVariable int batchId) {
+
```
add TODO comments

jurczakn commented on SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:
```diff
@@ -0,0 +1,239 @@
+	private MockMvc mockMvc;
+
+	@MockBean
+	private DistributionService service;
```
improve name

jurczakn commented on SurveyService/src/main/java/com/revature/service/DistributionServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+	 * TODO: Document after creating tests
+	 */
+	@Override
+	public List<String> sendEmailsByBatchIdAndCSV(int batchId, MultipartFile csv) {
```
have throw more specific exceptions
